azurerm/data_source_redis_cache.go

```diff
@@ -0,0 +1,266 @@
+
+			"capacity": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"family": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"minimum_tls_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"shard_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+
+			"enable_non_ssl_port": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"private_static_ip_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"redis_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"maxclients": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"maxmemory_delta": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+
+						"maxmemory_reserved": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+
+						"maxmemory_policy": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "volatile-lru",
+						},
+
+						"maxfragmentationmemory_reserved": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+
+						"rdb_backup_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+
+						"rdb_backup_frequency": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+
+						"rdb_backup_max_snapshot_count": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+
+						"rdb_storage_connection_string": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+						},
+
+						"notify_keyspace_events": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"aof_backup_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+
+						"aof_storage_connection_string_0": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+						},
+
+						"aof_storage_connection_string_1": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+						},
+						"enable_authentication": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+					},
+				},
+			},
+
+			"patch_schedule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"day_of_week": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"start_hour_utc": {
+							Type:     schema.TypeInt,
+							Optional: true,
```

Comment:
since this is readonly - can we make this Computed instead of Optional:

```suggestion
							Computed: true,
```